documentation/guide-batch-layer.asciidoc

```diff
@@ -7,791 +7,215 @@ We understand batch processing as bulk-oriented, non-interactive, typically long
-
-The date parameters can also be useful for controlling the business logic, e.g. a batch can process all data that was created until the current date (as set in the date parameter), thereby giving a chance to control how much is actually processed.
-
-If your batch has to run more than once a day you could easily adapt the concept of timestamps. If you are using an external batch scheduler, they often provide a counter for the execution and you might automatically pass this instead of the date parameter.
-
-[[performance-tuning]]
-=== Performance Tuning
-
-Most important for performance are of course the algorithms that you write and how fast (and scalable) these are, which is the same as for client processing. Apart from that, the performance of batches is usually closely related to the performance of the database system.
-
-If you are retrieving information from the database, you can have one complex query executed in the `ItemReader` (via a DAO) retrieving all the information needed for the current set of items, or you can execute further queries in the `ItemProcessor` (or `ItemWriter`) on a per item basis to retrieve further information.
-
-The first approach usually results in far more performance, because there is an overhead for every query being executed and this approach results in less queries being executed. Note that there is a tradeoff between performance and maintainability here. If you put everything into the query executed by an `ItemReader`, this query can get quite complex.
-
-Using cursors instead of pagination as described in the section on xref:itemreader[ItemReaders] can result in a better performance for the same reason: When using a cursor, the query is only executed once, when using pagination, the query is usually executed once per chunk. You could of course manually cache items, however this easily leads to a high memory consumption.
-
-Further possibilities for optimizations are query (plan) optimization and adding missing database indexes.
-
-=== Testing
-
-The Section link:guide-testing.asciidoc[Testing] covers how to unit and integration test in detail. Therefore we focus here on testing batches.
-
-In order for the unit test to run a batch job the unit test class must extend the `AbstractSpringBatchIntegrationTest` class. Annotation used to load the job's `ApplicationContext`:
-
-`@SpringBootTest(classes = {...})`: Indicates which JavaConfig classes (attribute `classes`) 
-`@ImportResource("classpath:../sample_BatchContext.xml") : Indicates XML files that contain the `ApplicationContext`. Use `@ContextConfiguration(...)` if Spring Boot is not used.
-
-[source,java]
-----
-public abstract class AbstractSpringBatchIntegrationTest extends AbstractComponentTest {..}
----- 
-
-[source,java]
-----
-@SpringBootTest(classes = { SpringBootBatchApp.class }, webEnvironment = WebEnvironment.RANDOM_PORT)
-@ImportResource("cla
```

Comment:
side note: AsciiDoc seems to thing you are writing French and considers `D'spring` as "D´spring"
Seems kind of a bug (maybe only on the limited github renderer) that this happens in code/monospace.
You can try using a code block (with 3 back-ticks) to prevent this.
If also not working there is IMHO nothing we can do about it.